localstack/aws/client.py

```diff
@@ -1,27 +1,74 @@
 """Utils to process AWS requests as a client."""
+import io
+import logging
+
 from botocore.model import OperationModel
 from botocore.parsers import create_parser as create_response_parser
 from werkzeug import Response
 
 from localstack.aws.api import ServiceResponse
 
+LOG = logging.getLogger(__name__)
+
+
+class _ResponseStream(io.RawIOBase):
+    """
+    Wraps a Response and makes it available as a readable IO stream.
+
+    Adapted from https://stackoverflow.com/a/20260030/804840
+    """
+
+    def __init__(self, response: Response):
+        self.iterator = response.iter_encoded()
+        self._buf = None
+
+    def readable(self):
+        return True
+
+    def readinto(self, b):
+        try:
+            upto = len(b)  # We're supposed to return at most this much
+            chunk = self._buf or next(self.iterator)
+            output, self._buf = chunk[:upto], chunk[upto:]
+            b[: len(output)] = output
+            return len(output)
+        except StopIteration:
+            return 0  # indicate EOF
+
 
 def parse_response(operation: OperationModel, response: Response) -> ServiceResponse:
     """
-    Parses an HTTP response object into an AWS response object using botocore.
+    Parses an HTTP Response object into an AWS response object using botocore. It does this by adapting the
+    procedure of ``botocore.endpoint.convert_to_response_dict`` to work with Werkzeug's server-side response object.
 
     :param operation: the operation of the original request
     :param response: the HTTP response object containing the response of the operation
     :return: a parsed dictionary as it is returned by botocore
     """
-    response_dict = {  # this is what botocore.endpoint.convert_to_response_dict normally does
+    # this is what botocore.endpoint.convert_to_response_dict normally does
+    response_dict = {
         "headers": dict(response.headers.items()),  # boto doesn't like werkzeug headers
         "status_code": response.status_code,
-        "body": response.data,
         "context": {
             "operation_name": operation.name,
         },
     }
 
+    if response_dict["status_code"] >= 300:
+        response_dict["body"] = response.data
+    elif operation.has_event_stream_output:
+        # TODO
+        LOG.warning(
+            "don't know how to parse an event stream output for %s.%s",
+            operation.service_model.service_name,
+            operation.name,
+        )
+        response_dict["body"] = response.data
```

Comment:
I just dig a bit into this:
- The parsers create an instance of the `EventStream` with `response['body']` in this case: [parsers.py#L348](https://github.com/boto/botocore/blob/0dfbdb9750404e94741086c6fc018626474ce09a/botocore/parsers.py#L348)
- The `EventStream` class in botocore expects this param to be a raw bytes stream:
[eventstream.py#L595](https://github.com/boto/botocore/blob/0dfbdb9750404e94741086c6fc018626474ce09a/botocore/eventstream.py#L595)

So I think we should just use the `_ResponseStream` here as well?

Reply:
cool, yea i added an additional `_RawStream` wrapper that exposes the interface botocore's `EventStream` object seems to expect. i'm too lazy to test this right now, so leaving a TODO for the next person :P